### PR TITLE
Shopping cart page

### PR DESCRIPTION
**Pull request by SStegniy**

### CODE REVIEW

**Review by v-skochko — APPROVED** [review]
Approved!

**Review by Vsumskiy — APPROVED** [review]

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+
+  private createFormGroup(): void {
+    this.personForm = new FormGroup({
+      firstName: new FormControl(),
+      lastName: new FormControl(),
+      email: new FormControl(),
+      phone: new FormControl(),
+      address: new FormControl(),
+      country: new FormControl(),
+      city: this.countriesFormControl,
```
Comment (maksym-syniuk) [inline figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:32]: Why `countryFormControl` for `city`?

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/countries-list.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
+export const countryList = [
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/countries-list.ts:1]: thanks, i will try to be more attentive

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 3]
```diff
@@ -1,15 +1,52 @@
...
 
 @Component({
   selector: 'app-shopping-cart',
   templateUrl: './shopping-cart.component.html',
   styleUrls: ['./shopping-cart.component.scss']
 })
 export class ShoppingCartComponent implements OnInit {
-
+  public personForm: FormGroup;
+  public countries = countryList;
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:14]: same answer

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 4]
```diff
@@ -1,15 +1,52 @@
...
+  public filteredCountries: Observable<string[]>;
+  public countriesFormControl = new FormControl();
   constructor() { }
 
   ngOnInit(): void {
+    this.createFormGroup();
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:20]: got it

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 5]
```diff
@@ -1,15 +1,52 @@
...
+    this.onFormChange();
+  }
+
+  private createFormGroup(): void {
+    this.personForm = new FormGroup({
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:25]: OK

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 6]
```diff
@@ -1,15 +1,52 @@
...
+      country: new FormControl(),
+      city: this.countriesFormControl,
+      postal: new FormControl(),
+    });
+  }
+
+  private onFormChange(): void {
+    this.filteredCountries = this.countriesFormControl.valueChanges.pipe(
+      startWith(''),
+      map(value => this.filterCountries(value))
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:40]: indicate to all the data type, got it

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 7]
```diff
@@ -1,15 +1,52 @@
...
+      phone: new FormControl(),
+      address: new FormControl(),
+      country: new FormControl(),
+      city: this.countriesFormControl,
+      postal: new FormControl(),
+    });
+  }
+
+  private onFormChange(): void {
+    this.filteredCountries = this.countriesFormControl.valueChanges.pipe(
```
[the remark this answers is not in the document]
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:38]: separate them

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts (hunk 1 above)
Comment (SStegniy) [reply figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:32]: that is a mistake, didn't notice how made it

File: figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts [hunk 8]
```diff
@@ -1,15 +1,56 @@
...
+      email: new FormControl(),
+      phone: new FormControl(),
+      address: new FormControl(),
+      country: this.countriesFormControl,
+      city: new FormControl(),
+      postal: new FormControl(),
+    });
+  }
+
+  private onFormChange(): any {
```
Comment (SStegniy) [inline figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts:38]: fix this method, return void

**Pull request merged**